### PR TITLE
Update structure, some names and imports

### PR DESCRIPTION
**Pull request by andrii-balitskyi**
Closes https://github.com/seamapi/python-next/issues/54 and https://github.com/seamapi/python-next/issues/55

### CODE REVIEW

**Comment by andrii-balitskyi** [comment]
@razor-x We have a circular import error with these changes. How do you think we should restructure the code to fix this?
image

**Comment by razor-x** [comment]
I don't know if this will fix things, but ideally the tests only need to import from the public API, e.g., instead of `from seam.exceptions import SeamApiException` it should be `from seam import SeamApiException`.

Meaning, importing anything like `seam.*` is NOT supported for consumers of this lib even if python allows it (can we prevent that usage?).

I understand if some unit tests need to import things that are not exposed on `seam`, however I can't think of any cases like that.

File: seam/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+from niquests import Response
+from .routes.models import ActionAttempt
+
+
+# HTTP
+class SeamApiException(Exception):
```
Comment (razor-x) [inline seam/exceptions.py:6]: I went back and forth on this name, but I landed on `SeamHttpApiError` in the JS SDK, and we should be consistent in the other projects. This gives us some space in the future to introduce errors from the Seam API that are not from the HTTP layer.

```suggestion
class SeamHttpApiError(Exception):
```

Comment (razor-x) [reply seam/exceptions.py:6]: Ah, there is already an issue for this :) https://github.com/seamapi/python-next/issues/48

Comment (andrii-balitskyi) [reply seam/exceptions.py:6]: Yep and [PR](https://github.com/seamapi/python-next/pull/58) is also ready!

**Comment by andrii-balitskyi** [comment]
@razor-x I fixed SeamApiException import in tests but circular import issue didn't go away. The issue is that seam/models.py imports AbstractRoutes and Workspace from seam.routes.models then seam/routes/__init__.py imports AccessCodes from seam.routes.access_codes and seam/routes/access_codes.py attempts to import AbstractSeam from seam.models, which is already being loaded, hence the circular import error.

**Review by razor-x — APPROVED** [review]

**Comment by andrii-balitskyi** [comment]
Closing in favour of https://github.com/seamapi/python-next/pull/63

**Pull request closed without merging**